## Task 1.1 Checkpoint 1 
### Socio-economical analysis of the customers.

In [3]:
import pandas as pd
import numpy as np
from pandas import Series,DataFrame

### Reading the Dataset related to Socio-Economic details of the customers

In [4]:
SocioEconomicData=pd.read_csv('Customer_social_economic_data_p1.csv') 

#### Counting the number of employees with Negative and Positive variation rate

In [5]:
Neg_Var_rate=SocioEconomicData[SocioEconomicData['emp.var.rate']<0].shape[0]
Pos_var_rate=SocioEconomicData[SocioEconomicData['emp.var.rate']>0].shape[0]

In [18]:
print("Among the total of ",Neg_Var_rate+Pos_var_rate,' employees')
print(f'The number of employess with negative variation rate is {Neg_Var_rate} which is {round(Neg_Var_rate*100/len(SocioEconomicData),2)} %\nThe number of employees with positive variation rate is {Pos_var_rate}  which is {round(Pos_var_rate*100/len(SocioEconomicData),2)} %')


Among the total of  37084  employees
The number of employess with negative variation rate is 15486 which is 41.76 %
The number of employees with positive variation rate is 21598  which is 58.24 %


In [16]:
pd.options.display.float_format='{:.1f}'.format
SocioEconomicData.describe()

emp.var.rate  cons.price.idx  cons.conf.idx  euribor3m  nr.employed
count       37084.0         37084.0        37084.0    37084.0      37084.0
mean            0.1            93.6          -40.5        3.6       5167.1
std             1.6             0.6            4.6        1.7         72.2
min            -3.4            92.2          -50.8        0.6       4963.6
25%            -1.8            93.1          -42.7        1.3       5099.1
50%             1.1            93.7          -41.8        4.9       5191.0
75%             1.4            94.0          -36.4        5.0       5228.1
max             1.4            94.8          -26.9        5.0       5228.1

#### The average consumer price index is 93.6 where the minimum is at 92.2 and maximum at 94.8
#### The average consumer confidence index is -40.5 where the minimum is at -50.8 and maximum at -26.9
#### The maximum interest for a customer is 5% and the minimum is 0.6%